### PR TITLE
Test train utils

### PR DESCRIPTION
**Pull request by tordbb**
Created a test for all functions inside of train_utils.py.

### CODE REVIEW

File: test/train/test_train_utils.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        self.assertEqual(self.image_generator.horizontal_flip, True)
+        self.assertEqual(self.image_generator.fill_mode, 'nearest')
+
+    def test_model(self):
+        # Test get_model output
+        self.assertTrue(self.model_was_built)
+
+    def test_train_model(self):
+        # Test train_model output
+        self.assertEqual(self.num_epochs, 25)
```
Comment (loeiten) [inline test/train/test_train_utils.py]: No need to train for 25 epochs as this will take time, 2 is more than enough

File: test/train/test_train_utils.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        # Run get_image_generator
+        self.image_generator = get_image_generator()
+
+        # Run get_model
+        model = get_model(len(set(labels)), epochs=1)
+        self.model_was_built = model._built
+
+        # Run train_model
+        history = train_model(model,
```
Comment (loeiten) [inline test/train/test_train_utils.py:111]: The `setUp` is for each member function of this class.
As there are 6 member functions you will train the same model with the same data and parameters 6 different times.
I would suggest to move the training to `test_train_model`

File: test/train/test_train_utils.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+from fruit_classifier.train.train_utils import train_model
+from fruit_classifier.train.train_utils import plot_training
+from fruit_classifier.preprocessing.preprocessing_utils import \
+    get_image_generator
+from pathlib import Path
+import shutil
+
+
+class TestTrainUtils(unittest.TestCase):
+    def setUp(self):
```
Comment (loeiten) [inline test/train/test_train_utils.py:15]: Recall that the `setUp` is common for all member functions.
If a variable is not shared with at least two member functions it should be refactored to the member function needing it.

For example:
Everything under
`# Run get_model_input`
is only needed in the `test_get_model_input` function, and should thus be moved there.

**Review by loeiten — APPROVED** [review]
Looks good, just needs some refactoring

**Review by loeiten — CHANGES_REQUESTED** [review]
Sorry, my fat fingers managed to click `Approve` while I meant clicking `Request changes`

File: test/train/test_train_utils.py (hunk 2 above)
Comment (tordbb) [reply test/train/test_train_utils.py:111]: But I will not be able to let the output from test_train_model(), and use it in test_outputs_were_right()? In that case I should also verify the outputs (the 6 tests) inside of test_train_model()?

File: test/train/test_train_utils.py (hunk 1 above)
Comment (tordbb) [reply test/train/test_train_utils.py]: Thanks, I needed to take a closer look at that!

File: test/train/test_train_utils.py (hunk 2 above)
Comment (tordbb) [reply test/train/test_train_utils.py:111]: I have now gathered the entire training, aswell as the testing of each output into one function.

File: test/train/test_train_utils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        shutil.copy(str(orig_file_path),
+                    str(self.folder_name.joinpath('B')))
+
+    def test_train_model(self):
+        # Run get_image_paths() and verify outputs
+        image_paths = get_image_paths(self.folder_name)
+        num_image_paths = len(image_paths)
+        self.assertGreater(num_image_paths, 0)
+
+        # Run get_data_and_labels and verify outputs
```
Comment (loeiten) [inline test/train/test_train_utils.py:53]: I see that I could have explained better.
I liked the way that you divided in functions previously, my only concern was that `setUp` was full of unrelated stuff

**Review by loeiten — CHANGES_REQUESTED** [review]
Good refactoring of `setUp`, but it's better to split in small functions. I'll refactor, push and merge

**Review by loeiten — APPROVED** [review]
